contracts/TokenProxy.sol

```diff
@@ -56,12 +53,9 @@ contract Gate {
 ///
 /// In the step 3 the User's tokens are going to be moved from the Gate to
 /// the User's balance in the Proxy.
-contract TokenProxy {
-
-    TransferableToken public TOKEN;
+contract TokenProxy is BasicToken {
```

Comment:
I guess the only thing left is to make it derive from ExtendedToken and we're done?

Reply:
I'm working on this.